libraries/AP_Airspeed/AP_Airspeed_MS4525.cpp

```diff
@@ -22,55 +22,78 @@
 #include <AP_HAL/AP_HAL.h>
 #include <AP_HAL/I2CDevice.h>
 #include <AP_Math/AP_Math.h>
+#include <GCS_MAVLink/GCS.h>
 #include <stdio.h>
 #include <utility>
 
 extern const AP_HAL::HAL &hal;
 
-#define MS4525D0_I2C_ADDR 0x28
+#define MS4525D0_I2C_ADDR1 0x28
+#define MS4525D0_I2C_ADDR2 0x36
+#define MS4525D0_I2C_ADDR3 0x46
 
 AP_Airspeed_MS4525::AP_Airspeed_MS4525(AP_Airspeed &_frontend, uint8_t _instance) :
     AP_Airspeed_Backend(_frontend, _instance)
 {
 }
 
-// probe and initialise the sensor
-bool AP_Airspeed_MS4525::init()
+// probe for a sensor
+bool AP_Airspeed_MS4525::probe(uint8_t bus, uint8_t address)
 {
-    const struct {
-        uint8_t bus;
-        uint8_t addr;
-    } addresses[] = {
-        { 1, MS4525D0_I2C_ADDR },
-        { 0, MS4525D0_I2C_ADDR },
-        { 2, MS4525D0_I2C_ADDR },
-        { 3, MS4525D0_I2C_ADDR },
-    };
-    bool found = false;
-    for (uint8_t i=0; i<ARRAY_SIZE(addresses); i++) {
-        _dev = hal.i2c_mgr->get_device(addresses[i].bus, addresses[i].addr);
-        if (!_dev) {
-            continue;
-        }
-        WITH_SEMAPHORE(_dev->get_semaphore());
+    _dev = hal.i2c_mgr->get_device(bus, address);
+    if (!_dev) {
+        return false;
+    }
+    WITH_SEMAPHORE(_dev->get_semaphore());
 
-        // lots of retries during probe
-        _dev->set_retries(10);
+    // lots of retries during probe
+    _dev->set_retries(10);
     
-        _measure();
-        hal.scheduler->delay(10);
-        _collect();
+    _measure();
+    hal.scheduler->delay(10);
+    _collect();
 
-        found = (_last_sample_time_ms != 0);
-        if (found) {
-            printf("MS4525: Found sensor on bus %u address 0x%02x\n", addresses[i].bus, addresses[i].addr);
-            break;
+    return _last_sample_time_ms != 0;
+}
+
+// probe and initialise the sensor
+bool AP_Airspeed_MS4525::init()
+{
+    const uint8_t addresses[] = { MS4525D0_I2C_ADDR1, MS4525D0_I2C_ADDR2, MS4525D0_I2C_ADDR3 };
+    if (bus_is_confgured()) {
+        // the user has configured a specific bus
+        for (uint8_t addr : addresses) {
```

Comment:
Wow, we really are going all-out with the new iterator syntax now :-)

Reply:
yes